docs/examples/tutorial_signal_processing.py

```diff
@@ -0,0 +1,397 @@
+        for data in tqdm.tqdm(
+            r.iter_content(block_size),
+            unit="MB",
+            unit_scale=True,
+            total=math.ceil(int(r.headers.get("content-length", 0)) // block_size),
+        ):
+            f.write(data)
+
+
+# %%
+# ***
+# Loading the data
+# ------------------
+# Let's load and print the full dataset.
+
+data = nap.load_file(path)
+FS = 1250
+print(data)
+
+
+# %%
+# ***
+# Selecting slices
+# -----------------------------------
+# We will consider a single run of the experiment - where the rodent completes a full traversal of the linear track,
+# followed by 4 seconds of post-traversal activity.
+
+# Define the run to use for this Analysis
+run_index = 7
+# Define the IntervalSet for this run and instantiate both LFP and
+# Position TsdFrame objects
+RUN_interval = nap.IntervalSet(
+    data["forward_ep"]["start"][run_index],
+    data["forward_ep"]["end"][run_index] + 4.0,
+)
+RUN_Tsd = data["eeg"].restrict(RUN_interval)
+RUN_pos = data["position"].restrict(RUN_interval)
+print(RUN_Tsd)
+
+# %%
+# ***
+# Plotting the LFP and Behavioural Activity
+# -----------------------------------
+
+fig = plt.figure(constrained_layout=True, figsize=(10, 6))
+axd = fig.subplot_mosaic(
+    [["ephys"], ["pos"]],
+    height_ratios=[1, 0.4],
+)
+
+axd["ephys"].plot(
+    RUN_Tsd[:, 0].restrict(
+        nap.IntervalSet(
+            data["forward_ep"]["start"][run_index], data["forward_ep"]["end"][run_index]
+        )
+    ),
+    label="Traversal LFP Data",
+    color="green",
+)
+axd["ephys"].plot(
+    RUN_Tsd[:, 0].restrict(
+        nap.IntervalSet(
+            data["forward_ep"]["end"][run_index],
+            data["forward_ep"]["end"][run_index] + 5.0,
+        )
+    ),
+    label="Post Traversal LFP Data",
+    color="blue",
+)
+axd["ephys"].set_title("Traversal & Post Traversal LFP")
+axd["ephys"].set_ylabel("LFP (v)")
+axd["ephys"].set_xlabel("time (s)")
+axd["ephys"].margins(0)
+axd["ephys"].legend()
+axd["pos"].plot(RUN_pos, color="black")
+axd["pos"].margins(0)
+axd["pos"].set_xlabel("time (s)")
+axd["pos"].set_ylabel("Linearized Position")
+axd["pos"].set_xlim(RUN_Tsd.index.min(), RUN_Tsd.index.max())
+
+
+# %%
+# ***
+# Getting the LFP Spectrogram
+# -----------------------------------
+# Let's take the Fourier transform of our data to get an initial insight into the dominant frequencies.
+
+fft = nap.compute_spectogram(RUN_Tsd, fs=int(FS))
+print(fft)
+
+# %%
+# ***
+# The returned object is a pandas dataframe which uses frequencies as indexes and spectral power as values.
+#
+# Now let's plot it
+
+fig, ax = plt.subplots(1, constrained_layout=True, figsize=(10, 4))
+ax.plot(
+    fft.index,
+    np.abs(fft.iloc[:, 0]),
+    alpha=0.5,
+    label="LFP Frequency Power",
+    c="blue",
+    linewidth=2,
+)
+ax.set_xlabel("Freq (Hz)")
+ax.set_ylabel("Frequency Power")
+ax.set_title("LFP Fourier Decomposition")
+ax.set_xlim(1, 30)
+ax.axvline(9.36, c="orange", label="9.36Hz", alpha=0.5)
```

Comment:
No need for this. Eventually i would do a axvspan from 8 to 12 if you want to emphasize theta.

Reply:
Have you tried loglog or semilogy?

Reply:
Some example are shown here : https://www.nature.com/articles/s41593-020-00744-x

Reply:
I have but they did not look good. I will show you in our next meeting.